flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/hashtable/BinaryHashTable.java

```diff
@@ -399,10 +406,37 @@ private boolean prepareNextPartition() throws IOException {
         this.probeMatchedPhase = true;
         this.buildIterVisited = false;
 
+        final int nextRecursionLevel = p.getRecursionLevel() + 1;
+        if (nextRecursionLevel == 2) {
+            LOG.info("Recursive hash join: partition number is " + p.getPartitionNumber());
+        } else if (nextRecursionLevel > MAX_RECURSION_DEPTH) {
+            LOG.info(
+                    String.format(
+                            "Partition number [%s] recursive level more than %s.",
+                            p.getPartitionNumber(), MAX_RECURSION_DEPTH));
```

Comment:
1. Use slf4j `{}` instead of `String.format`. 
2. Add "Process the partition using SortMergeJoin later" in the log?